test/ExecutionTests.jl

```diff
@@ -148,12 +148,26 @@ tune!(b_pass)
 # warmup #
 ###########
 
-p = params(warmup(@benchmarkable sin(1)))
+is_warm = false
+function needs_warm()
+    global is_warm
+    if is_warm
+        sleep(0.1)
+    else
+        sleep(2)
+        is_warm = true
+    end
+end
+
+w = @benchmarkable needs_warm()
+w.params.seconds = 1
+tune!(w)
+is_warm = false
+@test minimum(run(w).times) < 1e9
```

Comment:
After this `is_warm` I'd true right? If so does it make sense to test that.

Reply:
`is_warm` would be true even before merging this PR (it becomes true if the function has been called at least once), so checking whether it is true would not test the behavior we want to add. I'd be happy to add a counter to the function that tests the number of executions, which should be at least 2, if you like?

Reply:
Oh right sure, nevermind then.